src/profile.js

```diff
@@ -34,6 +35,35 @@ module.exports = function generateHeapSamplingProfile(options, cb) {
     throw new Error('The interval option must be a number greater than 0')
   }
 
+  if (signal) {
+    if (signal.addEventListener) {
```

Comment:
Check that signal has not already been tripped here.

Reply:
done